third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -875,6 +875,137 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
 /// # Definition Analysis
 
 impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
+    pub fn check_language_version(&self, loc: &Loc, feature: &str, version_min: LanguageVersion) {
+        if !self.parent.env.language_version().is_at_least(version_min) {
+            self.parent.env.error(
+                loc,
+                &format!(
+                    "not supported before language version `{}`: {}",
+                    version_min, feature
+                ),
+            )
+        }
+    }
+
+    fn analyze_constant(
```

Comment:
Can we have some documentation of this function?  Also, `analyze_constants`.  These names are painfully generic.

Reply:
done